### PR TITLE
end to end fixs

### PR DESCRIPTION
**Pull request by neildsouth**
Signed-off-by: Neil South <neil.south@answerdigital.com>

### Description

Fixes # .

A few sentences describing the changes proposed in this pull request.

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x ] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.

### CODE REVIEW

File: src/TaskManager/Plug-ins/Argo/ArgoProvider.cs [hunk 1]
```diff
@@ -27,26 +27,34 @@ namespace Monai.Deploy.WorkflowManager.TaskManager.Argo
...
     {
         private readonly ILogger<ArgoProvider> _logger;
-        private readonly IHttpClientFactory _httpClientFactory;
+        private readonly IHttpClientFactory _clientFactory;
 
         public ArgoProvider(ILogger<ArgoProvider> logger, IHttpClientFactory httpClientFactory)
         {
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
-            _httpClientFactory = httpClientFactory ?? throw new ArgumentNullException(nameof(httpClientFactory));
+            _clientFactory = httpClientFactory;
```
Comment (mocsharp) [inline src/TaskManager/Plug-ins/Argo/ArgoProvider.cs:35]: Any reason to remove the null check here?

File: src/TaskManager/TaskManager.cs [hunk 2]
```diff
@@ -323,21 +324,29 @@ private async Task HandleDispatchTask(JsonMessage<TaskDispatchEvent> message)
...
+                bucket: storage.Bucket,
+                payloadId: GetPayloadIdFormPath(storage.RelativeRootPath)))
+                    .Distinct()
+                    .ToArray();
 
             var creds = await _storageAdminService.CreateUserAsync(
                 $"TM{Guid.NewGuid()}",
-                 AccessPermissions.Read,
-                 storageBuckets);
+                 storageBuckets.Select(b => new PolicyRequest(b.bucket, b.payloadId)).ToArray()
```
Comment (mocsharp) [inline src/TaskManager/TaskManager.cs:348]: 😂 I have updated these in the other PR as well

File: src/TaskManager/Plug-ins/Argo/ArgoProvider.cs (hunk 1 above)
Comment (neildsouth) [reply src/TaskManager/Plug-ins/Argo/ArgoProvider.cs:35]: no, that was a mistake, will put it back

File: src/TaskManager/Runner/Program.cs [hunk 3]
```diff
@@ -60,7 +60,7 @@ private static async Task Main(string[] args)
             var wmConfig = host.Services.GetRequiredService<IOptions<WorkflowManagerOptions>>();
             Guard.Against.NullService(wmConfig, nameof(IOptions<WorkflowManagerOptions>));
 
-            subscriber.Subscribe(messagingKeys.TaskUpdateRequest, string.Empty, (args) =>
+            subscriber.Subscribe(messagingKeys.TaskUpdateRequest, "TaskUpdate", (args) =>
```
Comment (jackschofield23) [inline src/TaskManager/Runner/Program.cs:63]: I think this should be also messagingKeys.TaskUpdateRequest to keep it consistent

**Review by jackschofield23 — APPROVED** [review]

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 4]
```diff
@@ -97,19 +98,18 @@ public async Task<bool> ProcessPayload(WorkflowRequestEvent message, Payload pay
...
-            foreach (var workflow in workflows)
-            {
-                var workflowInstance = await CreateWorkflowInstanceAsync(message, workflow);
-                workflowInstances.Add(workflowInstance);
-            }
+            var tasks = workflows.Select(workflow => CreateWorkflowInstanceAsync(message, workflow));
+            await Task.WhenAll(tasks);
+            workflowInstances.AddRange(tasks.Select(t => t.Result));
 
+            Debug.Assert(workflowInstances.Count > 0);
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:112]: is this debugging code left in?

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 5]
```diff
@@ -97,19 +98,18 @@ public async Task<bool> ProcessPayload(WorkflowRequestEvent message, Payload pay
...
             }
 
             var workflowInstances = new List<WorkflowInstance>();
 
-            foreach (var workflow in workflows)
-            {
-                var workflowInstance = await CreateWorkflowInstanceAsync(message, workflow);
-                workflowInstances.Add(workflowInstance);
-            }
+            var tasks = workflows.Select(workflow => CreateWorkflowInstanceAsync(message, workflow));
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:108]: would this be simpler as 
            var tasks = await workflows.Select(async workflow => await CreateWorkflowInstanceAsync(message, workflow));
            
            I think is the syntax?

File: src/WorkflowManager/appsettings.Local.json [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      },
+      "meta-data": {
+        "argo": "Monai.Deploy.WorkflowManager.TaskManager.Argo.Repositories.ArgoMetadataRepository, Monai.Deploy.WorkflowManager.TaskManager.Argo",
+        "test": "Monai.Deploy.WorkflowManager.TaskManager.TestPlugin.Repositories.TestPluginRepository, Monai.Deploy.WorkflowManager.TaskManager.TestPlugin"
+      }
+    },
+    "storage": {
+      "serviceAssemblyName": "Monai.Deploy.Storage.MinIO.MinIoStorageService, Monai.Deploy.Storage.MinIO",
+      "settings": {
+        "endpoint": "minio:30090", // note edit C:\Windows\System32\drivers\etc\Hosts fand add  `127.0.0.1	minio`
```
Comment (lillie-dae) [inline src/WorkflowManager/appsettings.Local.json:29]: I dont think putting comments like this in json like this is valid.

Comment (lillie-dae) [reply src/WorkflowManager/appsettings.Local.json:29]: https://reqbin.com/json/5wzepqmt/json-comment-example#:~:text=Can%20I%20add%20a%20comment,elements%20will%20still%20be%20data.

**Review by lillie-dae — APPROVED** [review]

**Pull request merged**